torch/nn/_functions/thnn/auto.py

```diff
@@ -23,34 +26,54 @@ def _make_function_class_criterion(class_name, update_output, update_grad_input,
             buffers_idx.append(additional_arg_idx)
         additional_arg_idx += 1
 
-    def __init__(self, *args, **kwargs):
-        Function.__init__(self)
-        self.weight = kwargs.get('weight')
-        self.additional_args = list(args)
+    @staticmethod
+    def backward_cls_forward(ctx, input, target, grad_output, additional_args_ctx, backend_ctx):
```

Comment:
can you move `backward_cls_forward` and `backward_cls_backward` under `forward` and `backward`? This would make reviewing changes much easier. Right now git is trying to match the diff of backward class functions with the forward class functions